File: 0326_Power_of_Three/326.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n == 0:
            return False
        while True:
            if n == 1:
                return True
            if n % 3 == 0:
                n /= 3
            else:
                return False

a = Solution()
print(a.isPowerOfThree(n = 27))
print(a.isPowerOfThree(n = 0))
print(a.isPowerOfThree(n = 9))
print(a.isPowerOfThree(n = 45))

File: 0326_Power_of_Three/326.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        while (n > 1) {
            if (n % 3 == 0)
                n /= 3;
            else
                return false;
        }
        return n == 1;
    }
};

int main() {
    Solution a;
    cout << a.isPowerOfThree(27) << " " << a.isPowerOfThree(0) << "\n";
    return 0;
}
